model2vec/inference/model.py

```diff
@@ -305,3 +348,46 @@ def evaluate_single_or_multi_label(
     )
 
     return report
+
+
+def get_most_important_tokens(model: Any, text: str) -> list[tuple[str, float]]:
+    """
+    Get the most important tokens for the predicted class.
+
+    :param model: The model to get the most important tokens for.
+    :param text: The text to get the most important tokens for.
+    :return: A list of (token, score) tuples sorted by descending importance.
+    """
+    # Use the model's tokenizer to convert the text to token IDs
+    if isinstance(model, StaticModelPipeline):
+        input_ids = model.model.tokenize([text])
+    else:
+        input_ids = model.tokenize([text]).tolist()
+    unique_ids = set(input_ids[0])
+
+    # Identify tokens that are not yet cached and compute their logits
+    tokens_to_compute = [token_id for token_id in unique_ids if token_id not in model.token_logits_cache]
+    if tokens_to_compute:
+        computed_logits = model.predict_logits(token_ids=tokens_to_compute)
+        # Update the cache with the computed logits
+        for token_id, logit in zip(tokens_to_compute, computed_logits):
+            model.token_logits_cache[token_id] = logit
+
+    # Determine the predicted label for the text
+    probs = model.predict_proba([text])[0]
+    label_idx = int(np.argmax(probs))
+
+    results = []
+    for token_id in unique_ids:
+        # Get the token string and logit
+        token_str = model.tokenizer.id_to_token(token_id)
+        token_logit = model.token_logits_cache.get(token_id)
+        if token_logit is None:
+            continue
+        # Get the logit for the predicted label
+        score = float(token_logit[label_idx])
```

Comment:
This looks kind of odd to me. Why would the logit of the predicted label, and not the collected logits of the token over all classes, determine the classification. For example, consider a situation in which a token gets high logits for 2 out of many classes. In that case ,it might not be important at all, right?